src/ripple/app/misc/impl/AmendmentTable.cpp

```diff
@@ -326,31 +362,87 @@ AmendmentTableImpl::AmendmentTableImpl(
         }
     }
 
+    hash_set<uint256> detect_conflict;
+    // Parse enabled amendments from config
     for (auto const& a : parseSection(enabled))
     {
-        if (auto s = add(a.first, sl))
-        {
-            JLOG(j_.debug()) << "Amendment " << a.first << " is enabled.";
-
-            if (!a.second.empty())
-                s->name = a.second;
-
-            s->supported = true;
-            s->enabled = true;
+        if (FeatureVotesExist)
+        {  // If the table existed, warn about duplicate config info
+            JLOG(j_.warn()) << "[amendments] section in config file ignored"
+                               " in favor of data in db/wallet.db.";
+            break;
+        }
+        else
+        {  // Otherwise transfer config data into the table
+            detect_conflict.insert(a.first);
+            persistVote(a.first, a.second, false);  // un-veto
         }
     }
 
+    // Parse enabled amendments from config
     for (auto const& a : parseSection(vetoed))
     {
-        // Unknown amendments are effectively vetoed already
-        if (auto s = get(a.first, sl))
-        {
-            JLOG(j_.info()) << "Amendment " << a.first << " is vetoed.";
-
-            if (!a.second.empty())
-                s->name = a.second;
+        if (FeatureVotesExist)
+        {  // If the table existed, warn about duplicate config info
+            JLOG(j_.warn())
+                << "[veto_amendments] section in config file ignored"
+                   " in favor of data in db/wallet.db.";
+            break;
+        }
+        else
+        {  // Otherwise transfer config data into the table
+            if (detect_conflict.count(a.first) == 0)
+            {
+                persistVote(a.first, a.second, true);  // veto
+            }
+            else
+            {
+                JLOG(j_.warn())
+                    << "[veto_amendments] section in config has amendment "
+                    << '(' << a.first << ", " << a.second
+                    << ") both [veto_amendments] and [amendments].";
+            }
+        }
+    }
 
-            s->vetoed = true;
+    // Read amendment votes from wallet.db
+    auto db = db_.checkoutDb();
+    soci::transaction tr(*db);
+    std::string sql =
+        "SELECT AmendmentHash, AmendmentName, Veto FROM FeatureVotes";
+    boost::optional<std::string> amendment_hash;
+    boost::optional<std::string> amendment_name;
+    boost::optional<int> vote;
+    soci::statement st =
+        (db->prepare << sql,
+         soci::into(amendment_hash),
+         soci::into(amendment_name),
+         soci::into(vote));
+    st.execute();
+    while (st.fetch())
+    {
+        uint256 amend_hash = from_hex_text<uint256>(*amendment_hash);
+        if (*vote)  // if veto
```

Comment:
My intuition tells me that a variable named `vote` is going to tell me whether I should vote _for_ a thing. This would be significantly clearer if `vote` was renamed `vetoed` or `voteVeto` or something like that.

Reply:
Done.